supporter-product-data/cloudformation/cfn.yaml

```diff
@@ -0,0 +1,383 @@
+                      "NumericEquals": 0,
+                      "Next": "NoNewSubscriptions"
+                  }
+                ],
+                "Default": "UpdateDynamo"
+              },
+              "NoNewSubscriptions": {
+                "Type": "Pass",
+                "End": true
+              },
+              "UpdateDynamo": {
+                "Type": "Task",
+                "Resource": "${dynamoArn}",
+                "Next": "CheckProcessingComplete",
+                "Retry": [
+                   {
+                    "ErrorEquals": ["States.ALL"],
+                    "IntervalSeconds": 30,
+                    "MaxAttempts": 3,
+                    "BackoffRate": 1.0
+                  }]
+              },
+              "CheckProcessingComplete": {
+                "Type": "Choice",
+                "Choices": [
+                  {
+                      "Variable": "$.recordCount",
+                      "NumericGreaterThanPath": "$.processedCount",
+                      "Next": "UpdateDynamo"
+                  }
+                ],
+                "Default": "ProcessingComplete"
+              },
+              "ProcessingComplete": {
+                "Type": "Pass",
+                "End": true
+              }
+            }
+          }
+        - {
+          querierArn: !GetAtt [ QueryZuoraLambda, Arn ],
+          fetcherArn: !GetAtt [ FetchResultsLambda, Arn ],
+          dynamoArn: !GetAtt [ UpdateDynamoLambda, Arn ]
+        }
+      RoleArn: !GetAtt [ StatesExecutionRole, Arn ]
+
+  SupporterProductDataScheduledRule:
+    Type: "AWS::Events::Rule"
+    Properties:
+      Description: "Trigger regular data synchronisation from Zuora"
+      ScheduleExpression: !FindInMap [StageVariables, !Ref Stage, scheduleRate]
+      State: "ENABLED"
+      Targets:
+        -
+          Arn: !Ref SupporterProductData
+          Id: !Sub trigger_state_machine-${Stage}-1
+          Input: |
+            {
+              "queryType":"incremental"
+            }
+          RoleArn: !GetAtt [ SupporterProductDataTriggerRole, Arn ]
+
+  SupporterProductDataTriggerRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - events.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: TriggerStateMchine
+          PolicyDocument:
+            Version : "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - states:StartExecution
+                Resource: !Ref SupporterProductData
+
+  SupporterProductDataAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: CreateProdResources
+    Properties:
+      AlarmActions:
+        - !Sub arn:aws:sns:${AWS::Region}:${AWS::AccountId}:reader-revenue-dev
+      AlarmName: !Join
+        - ' '
+        - - 'URGENT 9-5 -'
```

Comment:
I'm not totally set either way, but I would say it's not "drop everything - it's down" urgent